content/kb/troubleshooting-rwx-encrypted-volume-fails-to-live-expand.md

```diff
@@ -0,0 +1,116 @@
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+...
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: "73400320"
+  storageClassName: longhorn-crypto-global
+  volumeMode: Filesystem
+  volumeName: pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+status:
+  accessModes:
+    - ReadWriteMany
+  capacity:
+    storage: 50Mi
+  conditions:
+    - lastProbeTime: null
+      lastTransitionTime: "2025-07-01T09:16:05Z"
+      message: Waiting for user to (re-)start a pod to finish file system resize of
+        volume on node.
+      status: "True"
+      type: FileSystemResizePending
+  phase: Bound
+```
+
+At the meantime, in the corresponding Longhorn share manager logs, it keep complaining "unsupported disk encryption format":
+
+```
+time="2025-07-01T09:29:42Z" level=error msg="Failed to resize mounted filesystem on volume" error="rpc error: code = InvalidArgument desc = unsupported disk encryption format ext4" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+time="2025-07-01T09:31:24Z" level=info msg="Resizing mounted volume" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+time="2025-07-01T09:31:24Z" level=info msg="Device /dev/mapper/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e contains filesystem of format ext4" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+time="2025-07-01T09:31:24Z" level=error msg="Failed to resize mounted filesystem on volume" error="rpc error: code = InvalidArgument desc = unsupported disk encryption format ext4" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+time="2025-07-01T09:31:44Z" level=info msg="Resizing mounted volume" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+time="2025-07-01T09:31:44Z" level=info msg="Device /dev/mapper/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e contains filesystem of format ext4" filesystem=/export/pvc-a01aa079-b05d-40ea-a556-5ba691f9191e volume=pvc-a01aa079-b05d-40ea-a556-5ba691f9191e
+```
+
+## Root Cause
+
+During Kubernetes volume live expansion, the CSI plugin node server asks Longhorn share manager to expand the storage capacity. After the capacity gets expanded, the share manager first checks if the underlayer device is a LUKS encrypted device, then resize the mapped device to sync the capacity. But the share manager actually checks the type of mapped device, hence CSI plugin node server keeps receiving resize failure and consider the expansion is in complete.
+
+## Mitigation
+
+### Option A. Scale Down The Workload And Fix The State
```

Comment:
Any potential side effect?

Reply:
After discussion, we prefer not to touch the PVC. Will keep the only way to help user upgrading Longhorn.